src/controllers/roles.js

```diff
@@ -0,0 +1,38 @@
+import Responses from '../utils/response';
+import db from '../models';
+import UserServices from '../services/User.service';
+
+/**
+ * Class for roles related operations
+ */
+class Role {
+  /**
+   *
+   * @param {object} req
+   * @param {object} res
+   * @param {function} next
+   * @returns {object} responses
+   */
+  async changeRole(req, res) {
+    const user = await UserServices.findUserByEmail(req.body.email);
+    if (!user) {
+      return Responses.handleError(404, 'user not found', res);
+    }
+    await db.user.update(
```

Comment:
Since we have a `UserService` here, I think we can move this functionality into the user service.

Reply:
Changing it will break the code in the User's Controller. The methods 'forgotPassword' and 'findUser' use it in slightly different ways.

Reply:
I meant create a new method in the UserService and make use it in the controller.

Reply:
I've tried to create the service like this:
```
  static async findUserByEmailWithResponse(email) {
    const user = await db.user.findOne({
      where: { email }
    });
    if (!user) return false;
    return true;
  }
```

And then edited the controller like this:
```
  async changeRole(req, res) {
    if (UserServices.findUserByEmailWithResponse(req.body.email)) {
      await db.user.update(
        {
          role: req.body.role
        },
        { where: { email: req.body.email } }
      );
      // refetch user from database with updated role
      const updatedUser = await UserServices.findUserByEmail(req.body.email);
      // build returned object here:
      const userObject = {
        firstName: updatedUser.firstName,
        lastName: updatedUser.lastName,
        role: updatedUser.role,
      };
      return Responses.handleSuccess(200, 'user role updated', res, userObject);
    }
    return Responses.handleError(404, 'user not found', res);
  }
```

But that's not working. How can I go about this?

Reply:
I was hoping you'd create a `setRole` method in the UserService and then move the role update part into it.

Reply:
I've created the setUserRole. Check and see if it's what you had in mind.

Reply:
I have now implemented this.